synapse/metrics/background_process_metrics.py

```diff
@@ -169,7 +189,41 @@ def update_metrics(self):
         )
 
 
-def run_as_background_process(desc: str, func, *args, bg_start_span=True, **kwargs):
+R = TypeVar("R")
+
+
+@overload
+def run_as_background_process(  # type: ignore[misc]
```

Comment:
What's being ignored here?

Reply:
It's to suppress `error: Overloaded function signatures 1 and 2 overlap with incompatible return types  [misc]`. I shall add a comment

I initially tried a `Union` but ran into a bug/deficiency in mypy where it couldn't figure out what `R` was at every call site.
https://github.com/matrix-org/synapse/pull/10847#discussion_r711249811

Reply:
I tried to improve the comments further